src/vmm/src/guest_config/aarch64/mod.rs

```diff
@@ -24,56 +25,62 @@ pub struct CpuConfiguration {
 
 impl CpuConfiguration {
     /// Creates new guest CPU config based on the provided template
-    pub fn new(vcpu: &VcpuFd, template: &Option<CpuTemplateType>) -> Result<Self, Error> {
-        match template {
-            Some(ref cpu_template) => Self::with_template(vcpu, cpu_template),
-            None => Ok(Self::default()),
-        }
-    }
-
-    /// Creates new guest CPU config based on the provided template
-    fn with_template(vcpu: &VcpuFd, template: &CpuTemplateType) -> Result<Self, Error> {
-        match template {
-            CpuTemplateType::Custom(template) => Self::with_applied_template(vcpu, template),
-            CpuTemplateType::Static(StaticCpuTemplate::V1N1) => {
-                let template = static_cpu_templates::v1n1::v1n1();
-                Self::with_applied_template(vcpu, &template)
-            }
-            _ => unreachable!("Options other than V1N1 are invalid"),
-        }
-    }
-
-    /// Creates new guest CPU config based on the provided template
-    fn with_applied_template(vcpu: &VcpuFd, template: &CustomCpuTemplate) -> Result<Self, Error> {
-        let regs = template
-            .reg_modifiers
+    pub fn new(_vm: &Vm, vcpu: &KvmVcpu, template: &CustomCpuTemplate) -> Result<Self, Error> {
```

Comment:
nitpick: I agree with the consistency of the constructor parameters being the same across x86 and aarch64. Should more of the impl CpuConfiguration be refactored to be up a module level to reduce duplicated code?

Reply:
Sorry, I didn't get your point. Which part do you think we can reduce?

Both x86 and aarch64 has already shared `impl CpuConfiguration`. Does that mean defining a trait for it? If so, that makes sense to me!

Reply:
I like the idea of introducing traits here personally (for apply_template type functions especially), but not really needed right now. What I was referring to short term is the `pub fn new(_vm: &Vm, vcpu: &KvmVcpu, template: &CustomCpuTemplate)` definition being moved to `guest_config/mod.rs` (something ISA independent). Don't worry too much about this point for now.